paw-webapp/persistence/src/main/java/ar/edu/itba/paw/webapp/persistence/CommentHibernateDao.java

```diff
@@ -89,6 +91,59 @@ public Comment reply(Comment comment, String body, User replier) {
 
 
     @Override
+    public Map<Thread, Long> countComments(Collection<Thread> threads) {
+        if (threads == null) {
+            throw new IllegalArgumentException();
+        }
+        if (threads.isEmpty()) {
+            return new HashMap<>(); // Avoids querying
+        }
+
+        // Used for easily getting an entity by its id.
+        final Map<Long, Thread> ids = threads.stream()
+                .collect(Collectors.toMap(Commentable::getId, Function.identity()));
+        //noinspection unchecked, JpaQlInspection
+        final List<Object[]> comments = em.createQuery("SELECT comment.thread.id, count(comment)" +
+                " FROM Comment comment" +
+                " WHERE comment.parentComment is null AND comment.thread IN :threads" +
+                " GROUP BY comment.thread.id")
+                .setParameter("threads", threads)
+                .getResultList();
+        // The comments list holds Object arrays with two elements each: comment id and likes count
+        final Map<Thread, Long> result = comments.stream()
+                .collect(Collectors.toMap(each -> ids.get((Long) each[0]), each -> (Long) each[1]));
+        threads.forEach(thread -> result.putIfAbsent(thread, 0L));
+        return result;
+    }
+
+
+    @Override
+    public Map<Comment, Long> countReplies(Collection<Comment> comments) {
+        if (comments == null) {
+            throw new IllegalArgumentException();
+        }
+        if (comments.isEmpty()) {
+            return new HashMap<>(); // Avoids querying
+        }
+
+        // Used for easily getting an entity by its id.
+        final Map<Long, Comment> ids = comments.stream()
+                .collect(Collectors.toMap(Commentable::getId, Function.identity()));
+        //noinspection unchecked, JpaQlInspection
+        final List<Object[]> likes = em.createQuery("SELECT comment.parentComment.id, count(comment)" +
+                " FROM Comment comment" +
+                " WHERE comment.parentComment IN :comments" +
+                " GROUP BY comment.parentComment.id")
+                .setParameter("comments", comments)
+                .getResultList();
+        // The likes list holds Object arrays with two elements each: entity id and likes count
```

Comment:
Copy pasta: holds Object arrays with two elements each: entity id and ~likes~ reply count